routes/api.py

```diff
@@ -583,3 +583,73 @@ def exit_api():
         raise RuntimeError("Not running with the Werkzeug Server")
     shutdown_func()
     return "The server has shut down"
+
+
+@routes.route("/register_query_data.api", methods=["POST"])
+def register_query_data_api():
```

Comment:
Looks like a security hole - anybody could call this with any client id?

Reply:
Yes, I'll extend the key functionality implemented for the speech API to this endpoint as well.